corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py

```diff
@@ -107,24 +116,72 @@ def migrate_from_migration(cls, apps, schema_editor):
     def couch_db(cls):
         return couch_config.get_db(cls.couch_db_slug())
 
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--verify-only',
+            action='store_true',
+            dest='verify_only',
+            default=False,
+            help="""
+                Don't migrate anything, instead check if couch and sql data is identical.
+                Only works for migrations that use a couch_id on the sql model.
+            """,
+        )
+        parser.add_argument(
+            '--skip-verify',
+            action='store_true',
+            dest='skip_verify',
+            default=False,
+            help="""
+                Migrate even if verifcation fails. This is intended for usage only with
+                models that don't support verification.
+            """,
+        )
+
     def handle(self, **options):
+        verify_only = options.get("verify_only", False)
+        skip_verify = options.get("skip_verify", False)
+
+        self.doc_count = get_doc_count_by_type(self.couch_db(), self.couch_doc_type())
+        self.diff_count = 0
+        self.doc_index = 0
 
-        doc_count = get_doc_count_by_type(self.couch_db(), self.couch_doc_type())
         logger.info("Found {} {} docs and {} {} models".format(
-            doc_count,
+            self.doc_count,
             self.couch_doc_type(),
             self.sql_class().objects.count(),
             self.sql_class().__name__,
         ))
-        doc_index = 0
         for doc in get_all_docs_with_doc_types(self.couch_db(), [self.couch_doc_type()]):
-            doc_index += 1
-            logger.info("Looking at {} doc #{} of {} with id {}".format(
-                self.couch_doc_type(),
-                doc_index,
-                doc_count,
-                doc["_id"]
-            ))
-            with transaction.atomic():
-                model, created = self.update_or_create_sql_object(doc)
-                logger.info("{} model for doc with id {}".format("Creating" if created else "Updated", doc["_id"]))
+            self.doc_index += 1
+            if not skip_verify:
+                self._verify_doc(doc)
+            if not verify_only:
+                self._migrate_doc(doc)
```

Comment:
Since the verify is done before migration (and verification is skipped if the SQL model doesn't exist), this will not catch the type of issue that motivated this PR, right?

This arrangement will be useful if you run a migration and then immediately run a verify after that, which is good, but maybe we can do better?

Since it's impossible to detect that type of problem until some damage has been done to persistent data, I thought you were hoping to do something like verify each document _after_ it is migrated and abort the migration early if bad data is detected. Did I misunderstand how you're hoping to make that work?

Reply:
Nope, my bad, this is also why the test is failing.

Reply:
fixed in 5b6ef56cf6d5e80fd87d5cb0d5e1e7b190883c70